packages/api-cli/src/api.ts

```diff
@@ -187,15 +201,30 @@ async function makeTx ({ fn, log }: CallInfo): Promise<void> {
   assert(CRYPTO.includes(sign), `The crypto type can only be one of ${CRYPTO.join(', ')} found '${sign}'`);
 
   const keyring = new Keyring();
-  const account = keyring.createFromUri(seed, {}, sign as 'ed25519');
+  let auth = keyring.createFromUri(seed, {}, sign as 'ed25519');
+
+  let signable;
+  if (sudo) {
+    const provider = new WsProvider(ws);
+    const api = await ApiPromise.create({ provider });
+    const adminId = await api.query.sudo.key();
+    // attempt to get the admin key. This should only be in the keyring if the user
+    // has provided the superuser's seed. We still attempt to fetch the admin ID's
+    // keypair; if that fetch fails, then we can fail immediately instead of waiting
+    // for the tx to be rejected.
+    auth = keyring.getPair(adminId.toString());
```

Comment:
Would actually just forgo all of this - the seed that they supply should be the correct sudo seed (as per all other transactions).

You could still retrieve the sudo key as a pre-check, i.e. 

`assert(adminId.eq(auth.address), 'Supplied seed does not match on-chain sudo key');`